game-core/src/test/java/games/strategy/engine/data/CompositeChangeMatcher.java

```diff
@@ -0,0 +1,42 @@
+package games.strategy.engine.data;
+
+import java.util.Arrays;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+
+@AllArgsConstructor
+public class CompositeChangeMatcher extends ChangeMatcher<CompositeChange> {
```

Comment:
When we get checkstyle upgraded to 8.3x, type javadocs will I think become required again. Probably best to try and add them proactively so we don't have to do as much to migrate.

Beyond checkstyle, a javadoc here with some example usages I think would be really helpful. Some sample code on how to use this would make it really easy if someone were want to pick this up, they could copy/paste it and modify it to fit their needs.